src/BenchmarksApps/Kestrel/PlatformBenchmarks/SocketPipes/SocketPipeWriter.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PlatformBenchmarks
+{
+    internal sealed class SocketPipeWriter : PipeWriter
+    {
+        // the biggest response that we can create for TechEmpower Plaintext is around 1200 bytes
+        // it's simply cheaper to allocate small array and reuse it compared to pooling
+        private const int BufferSize = 2 * 1024;
+
+        private Socket _socket;
+        private byte[] _array;
+        private int _offset;
+
+        public SocketPipeWriter(Socket socket)
+        {
+            _socket = socket;
+            _array = new byte[BufferSize];
+            _offset = 0;
+        }
+
+        public override void Advance(int bytes)
+        {
+            _offset += bytes;
+
+            if (_offset >= _array.Length)
```

Comment:
How does this happen? This would be a bug right? You can't Advance more than we gave you in GetMemory